lib/PerlIO.pm

```diff
@@ -92,107 +140,119 @@ refuse to be pushed on top of itself.
 It currently does I<not> mimic MS-DOS as far as treating of Control-Z
 as being an end-of-file marker.
 
-Based on the C<:perlio> layer.
-
-=item :utf8
-
-Declares that the stream accepts perl's I<internal> encoding of
-characters.  (Which really is UTF-8 on ASCII machines, but is
-UTF-EBCDIC on EBCDIC machines.)  This allows any character perl can
-represent to be read from or written to the stream. The UTF-X encoding
-is chosen to render simple text parts (i.e.  non-accented letters,
-digits and common punctuation) human readable in the encoded file.
+On DOS/Windows like architectures where this layer is part of the defaults,
+it also acts like the C<:perlio> layer, and removing the CRLF translation
+(such as with C<:raw>) will only unset the CRLF translation flag.  Since
+Perl 5.14, you can also apply another C<:crlf> layer later, such as when
+the CRLF translation must occur after an encoding layer.  On other
+architectures, it is a mundane CRLF translation layer and can be added and
+removed normally.
 
-(B<CAUTION>: This layer does not validate byte sequences.  For reading input,
-you should instead use C<:encoding(UTF-8)> instead of bare C<:utf8>.)
+    # translate CRLF after encoding on Perl 5.14 or newer
+    binmode $fh, ":raw:encoding(UTF-16LE):crlf"
+      or die "binmode failed: $!";
 
-Here is how to write your native data out using UTF-8 (or UTF-EBCDIC)
-and then read it back in.
+=item :utf8
 
-	open(F, ">:utf8", "data.utf");
-	print F $out;
-	close(F);
+Pseudo-layer that declares that the stream accepts Perl's I<internal>
+upgraded encoding of characters, which is approximately UTF-8 on ASCII
+machines, but UTF-EBCDIC on EBCDIC machines.  This allows any character
+Perl can represent to be read from or written to the stream.
 
-	open(F, "<:utf8", "data.utf");
-	$in = <F>;
-	close(F);
+This layer (which actually sets a flag on the preceding layer, and is
+implicitly set by any C<:encoding> layer) does not translate or validate
+byte sequences.  It instead indicates that the byte stream will have been
+arranged by other layers to be provided in Perl's internal upgraded
+encoding, which Perl code (and correctly written XS code) will interpret
+as decoded Unicode characters.
 
+B<CAUTION>: Do not use this layer to translate to or from UTF-8 bytes,
```

Comment:
IMHO using it to translate to UTF-8 is acceptable. Perl is unlikely to generate invalid bytes sequences. Though double-checking is not a bad thing either.

Reply:
How is the updated warning?